### DISCUSSION TITLE
Building a Name Resolution Service in Soroban

### DISCUSSION THREAD

**Discussion by tosinshada** [stellar/sorobounty-spectacular#19] [General]
Hi,

I would like to share the second part of a series of decentralized Applications built on Soroban. This project aims to build a Name Resolution Service on Soroban

**Codebase**: The codebase is available [here](https://github.com/TosinShada/tide-soroban-contract) with the smart contracts now in the sns-registry, sns-registrar and sns-resolver folders
**Tutorial**: The tutorial can be found [here](https://dev.to/shada/soroban-series-2-building-a-name-resolution-service-18da)
**Video Walkthrough**: I also did a video walkthrough of both the code base and the application which can be found [here](https://www.youtube.com/watch?v=GVZm3NicDKc)
**Demo**: The current version of the application is hosted [here](https://tide-sns-frontend.vercel.app/)

**Comment by briwylde08** [comment]
Hello there! Thank you so much for submitting to Sorobounty Spectacular! We’ll review your submission and be in contact with you within the next week. 🎉

**Comment by Julian-dev28** [comment]
Hey @TosinShada thanks for the submission. This would be a cool feature to see on Soroban!

It seems like there are some errors occurring with the demo. Are you able to ensure that it is working properly?



A success message and a failure message are both generated when registering a name.
<img width="1112" alt="Screenshot 2023-09-07 at 3 53 16 PM" src="https://github.com/stellar/sorobounty-spectacular/assets/73849597/fe5599ff-f396-4452-993e-d8a798853ac0">

Resolving name fails:
<img width="1128" alt="Screenshot 2023-09-07 at 3 53 30 PM" src="https://github.com/stellar/sorobounty-spectacular/assets/73849597/f7abef95-05f8-4e8e-8dd1-5eaab56c71eb">

Thanks in advance!

**Reply by tosinshada** [reply]
Hi Julian, kindly try again. I have updated the contracts

**Reply by Julian-dev28** [reply]
Nicely done!

It is working on my end

**Reply by Julian-dev28** [reply]
Hey @TosinShada 

Im going through building the dapp now and am having some trouble finding the setup process

I see scripts in the backend/frontend/nameservice but it is unclear when to run which scripts to get the dapp stood up.

The readme in the backend tells the user to run yarn clean but there is no script for `clean` in the package.json file of the backend. The same goes for `yarn setup`

Are you able to update the readme and provide clearer instructions for deploying the dapp?

Thanks in advance!

**Reply by tosinshada** [reply]
Hi Julian, this has been updated

**Reply by Julian-dev28** [reply]
Hey @TosinShada 

The main readme to follow along is not clear on how to set up the sns dapp. 

If you can provide clear instructions on how to set the sns dapp up point it at the newly deployed contract etc
that would be great.

**Reply by tosinshada** [reply]
Hey Julian, this has been done

**Reply by Julian-dev28** [reply]
Okay great! Thanks for updating.

I followed the directions in the readme


> ### Frontend
> Follow the steps beloww to setup the frontend
> ```bash
> cd frontend
> ```
> 
> #### Run the frontend
> You need to have deployed the smart contracts first. See above for more details
> ```bash
> yarn dev
>```

and this was the frontend

<img width="1476" alt="Screenshot 2023-09-12 at 12 20 08 PM" src="https://github.com/stellar/sorobounty-spectacular/assets/73849597/d63639f5-149f-41d8-96ed-666899c2236b">


I then navigated to the `nameservice` directory and ran `yarn dev` again and was able to get the correct front end.

Please specify in the Readme what the proper workflow is for this submission.

After this update, please ping me so I can do the final review and pass along to Bri.


Other than that, the dapp looks solid and I am happy to have reviewed it!

**Reply by tosinshada** [reply]
Hi Julian, thanks for pointing this out. I have updated the readme with the proper workflow.

**Reply by Julian-dev28** [reply]
Perfect

I'm going to do a final review and Bri will follow up with you after!

Again, thanks for this submission. It is awesome to see this kind of use case built with Soroban!

**Comment by briwylde08** [comment]
Hello @TosinShada! We've reviewed your submission! [🎉](https://emojipedia.org/party-popper/)

Given our guidelines, your submission may qualify for an award of $1,500 worth of XLM. To receive that award, please fill out [this form](https://stellarform.typeform.com/to/HBjEyCJt) completely and accurately.

Thanks again! Please let us know if you have any questions.

Best,
Bri

**Reply by tosinshada** [reply]
Hi Bri, thanks for this. I just wanted to confirm if my previous documentation done [here ](https://github.com/stellar/sorobounty-spectacular/discussions/9) will suffice.

Also, I'd like to confirm if submitting an updated version with more features for the ongoing Hackathon is possible.

**Reply by briwylde08** [reply]
Heya! I totally missed that you had submitted a project earlier! Yes, that information works great.

As for the Hackathon, yes, you can definitely submit a project you started for Sorobounty Spectacular to Sorobanathon: Road to Mainnet!
